### DISCUSSION TITLE
About the checkpoint file.

### DISCUSSION THREAD

**Discussion by suzukimain** [huggingface/diffusers#9559] [Q&A] [answered]
Hello. 
I have a rudimentary question,
Could you please tell me the exact names of the checkpoints that can be loaded by `from_pretrained` and `from_single_file`, respectively?
I also checked `convert_diffusers_to_original_stable_diffusion.py` and others, but I couldn't find much.
Thank you.

https://github.com/huggingface/diffusers/blob/8e7d6c03a366fdb0f551ce7b92f0871c863d4e08/scripts/convert_diffusers_to_original_stable_diffusion.py#L1

**Comment by asomoza** [comment]
Hi, you don't say for which model type you're referring. `from_pretrained` it's for loading multifolder diffusers format checkpoint, so whenever you see a model that has multiple directories inside like `vae` or `text_encoder` you can use the `from_pretrained`. You can also search in the hub for a specific pipeline, for example: https://huggingface.co/models?other=diffusers%3AStableDiffusionPipeline

`from_single_file` is for loading, as the name says, a single file checkpoint which is the format that the webuis use, an example of this would be any model that you download from civitai. So whenever you encounter a model that only has one unique single file, you can use this method to load it.

**Reply by suzukimain** [reply]
Sorry. I did not explain well enough.
I would like to know how to describe each file type in the README.
For example, I could write `Stable Diffusion checkpoint` for the file to be loaded in `from_single_file`.
On the other hand, I could not figure out how to describe the model format to be read in `from_pretrained`.

**Reply by asomoza** [reply, marked as answer]
Both are the same, both are `Stable Diffusion checkpoints`, one has all the modules inside one file and the other keeps them separated in subdirectories. Depends on the model architecture but for example, for a SD 1.5 both have a text_encoder, a vae and a unet.

As I told you, there's a lot of model architectures so the `from_pretrained` varies from model to model, you can read what each pipeline requires in the corresponding pipeline class, for example for SDXL is [here](https://github.com/huggingface/diffusers/blob/8e7d6c03a366fdb0f551ce7b92f0871c863d4e08/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L185)

**Reply by suzukimain** [reply]
I see. Thank you for your time.
